pkg/snapshot/store.go

```diff
@@ -788,8 +797,16 @@ func GetGlobalStore(ctx context.Context, kotsadmNamepsace string, kotsadmVeleroB
 
 func mapAWSBackupStorageLocationToStore(kotsadmVeleroBackendStorageLocation *velerov1.BackupStorageLocation, store *types.Store) error {
 	endpoint, isS3Compatible := kotsadmVeleroBackendStorageLocation.Spec.Config["s3Url"]
-
-	if !isS3Compatible {
+	var domainName string
+	u, err := url.Parse(endpoint)
+	if err == nil {
+		splitHost := strings.Split(u.Hostname(), ".")
+		if len(splitHost) > 2 {
+			domainName = splitHost[len(splitHost)-2]
+		}
+	}
+	// if it is not s3 compatible, check if the endpoint domain matches amazonaws
+	if !isS3Compatible || domainName == "amazonaws" {
```

Comment:
Instead of splitting host name, do this `strings.HasSuffix(host, ".amazonaws.com")`